BackgroundOperationComparison.md

```diff
@@ -0,0 +1,44 @@
+**Purpose of this document:**
+
+This document is intended as a high level survey of what background capabilities are allowed by operating systems and browsers.
+
+**Operating Systems**
+
+All of the major desktop operating systems allow native applications to do significant background work, limited only by scheduling algorithms that prevent starvation of other processes.
+
+Android has historically had a very open policy for background work, allowing any installed app to do background processing and use network resources. However, that caused extreme battery life and network bandwidth problems for users. In recent releases the OS has started being more restrictive of background operations. In Android Marshmallow, Doze mode was introduced, during which time background processing was limited, and in Nougat, Doze mode has been expanded to be much more aggressive. Additionally, Android is moving to a more controlled model with expansions to the Job Scheduler in Nougat, which gives apps a more restricted set of background operation triggers.
+
+While Android was historically permissive and is now becoming more restrictive, iOS has followed the opposite path. iOS has always limited the types of apps that can do background processing, although it has expanded those types over time using the Background Execution framework. Particularly interesting is that in iOS7, the ability to respond to push messages and run for 30 seconds was added. Since then, more types of apps have been allowed to run background tasks, and more system services have been developed to allow apps to hand off tasks such as large downloads to the system.
+
+**References**
+Android Doze Mode: 
+https://developer.android.com/training/monitoring-device-state/doze-standby.html
+Android Job Scheduler: 
+https://developer.android.com/preview/features/background-optimization.html
+iOS Background Execution:
+https://developer.apple.com/library/ios/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/BackgroundExecution/BackgroundExecution.html
```

Comment:
Could you make a Markdown-styled list of these? Currently it looks like this:

https://github.com/jenharkness/budget-api/blob/0a7bbdb215a5419a57c905152de372d4e54bc853/BackgroundOperationComparison.md